### ISSUE TITLE
Implementing new methods which treat incoming and rehearsal data separately

### ISSUE THREAD

**Issue by pclucas14** [ContinualAI/avalanche#643]
Hi,

I'm trying to integrate [a new method ](https://arxiv.org/pdf/2104.05025.pdf) in your wonderful library. Our method treats incoming and rehearsal data differently. We apply logit masking on the incoming data and process the rehearsal data as is commonly done. Looking at the current code, my understanding is the following : 

1) Replay plugins overwrite the dataloader by outputting samples from the concatenation of buffer and incoming data
2) In this current setup, the strategy has no obvious way of telling the two data streams apart, (expect by looking directly at the label values)

In your opinion, what would be the easiest way to design a new strategy which could receive data from the incoming stream and the buffer _as two separate streams ?_ . My goal is to favor code reusability and hopefully piggyback on what is already implemented. 

Thank you!

**Comment by vlomonaco** [comment]
Hi @pclucas14! I'm moving this to the Q&A discussion tab, as it seems more related to a usability question rather than a software issue. Let me know in case you want to move it back in the issues.
